marklogic-data-hub-central/ui/src/components/entities/mapping/source-entity-map/source-to-entity-map.test.tsx

```diff
@@ -37,7 +38,225 @@ describe('RTL Source-to-entity map tests', () => {
+
+      //Reset the search filter on Source table
+      fireEvent.click(filterIcon);
+      let resetSearch = getByTestId('ResetSearch-key');
+      expect(resetSearch).toBeInTheDocument();
+      fireEvent.click(resetSearch);
+
+      //Check if the expected values are present now after resetting the filter on source table.
+      expect(getByText('proteinId')).toBeInTheDocument();
+      expect(getByText('@proteinType')).toBeInTheDocument();
+      
+
+      /* Test filter on Entity table  */
+      let entityfilterIcon = getByTestId('filterIcon-name');
+      expect(entityfilterIcon).toBeInTheDocument();
+      fireEvent.click(entityfilterIcon);
+      let inputSearchEntity = getByTestId('searchInput-name');
+      expect(inputSearchEntity).toBeInTheDocument();
+      fireEvent.change(inputSearchEntity, { target: {value: "craft" }}); //Enter a case-insensitive value in inputEntitySearch field
+      expect(getByTestId('submitSearch-name')).toBeInTheDocument();
+      expect(inputSearchEntity).toHaveValue('craft');
+      fireEvent.click(getByTestId('submitSearch-name')); //Click on Search button to apply the filter with the desired string
+
+      //Check if the expected values are available/not available in search result.
+      expect(getByText('items')).toBeInTheDocument();
+      expect(getByText('itemTypes')).toBeInTheDocument();
+      expect(getByText('itemCategory')).toBeInTheDocument();
+      expect(getByText('Craft')).toBeInTheDocument();
+      expect(queryByText('propId')).not.toBeInTheDocument(); 
+      expect(queryByText('propName')).not.toBeInTheDocument();
+
+      //Check if the source table properties are not affected by the filter on Entity table
+      expect(getByText('proteinId')).toBeInTheDocument();
+      expect(getByText('@proteinType')).toBeInTheDocument();
+      expect(getByText('FirstNamePreferred')).toBeInTheDocument();
+      expect(getByText('LastName')).toBeInTheDocument();
+
+       //Reset the search filter on Entity table
+       fireEvent.click(entityfilterIcon);
+       let resetEntitySearch = getByTestId('ResetSearch-name');
+       expect(resetEntitySearch).toBeInTheDocument();
+       fireEvent.click(resetEntitySearch);
+ 
+       //Check if the expected values are present now after resetting the filter on Entity table.
+       expect(getByText('propId')).toBeInTheDocument();
+       expect(getByText('propName')).toBeInTheDocument();
+    });
+
+    test('Column option selector in Entity table',() => {
+      
+      const { getByText,getByTestId,queryByText } = render(<SourceToEntityMap {...data.mapProps} 
+        mappingVisible={true} 
+        entityTypeProperties={data.mapProps.entityTypePropertiesUpdated} 
+        sourceData={data.mapProps.sourceDataUpdated}/>);
+
+      //Set the data for texting in xpath expression
+
+      let exp = getByText('testNameInExp');
+      expect(exp).toBeInTheDocument();
+      fireEvent.change(exp, { target: {value: "concat(propName,'-NEW')" }});
```

Comment:
fireEvent on exp would fail if it was not found in the DOM. So we can get rid of the assertion in line 131 and other similar assertions like this in the test, since they wont add any value to the test.

Reply:
Done